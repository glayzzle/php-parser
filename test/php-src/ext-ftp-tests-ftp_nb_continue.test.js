// eslint-disable prettier/prettier
const parser = require("../main");

describe("php-src tests", function () {
  // ext/ftp/tests/ftp_nb_continue.phpt
  it("Testing whether ftp_nb_continue() fetches more data", function () {
    expect(parser.parseCode("<?php\nrequire 'server.inc';\n$file = \"mediumfile.txt\";\n$ftp = ftp_connect('127.0.0.1', $port);\nftp_login($ftp, 'user', 'pass');\nif (!$ftp) die(\"Couldn't connect to the server\");\n$local_file = __DIR__ . DIRECTORY_SEPARATOR . $file;\ntouch($local_file);\n$r = ftp_nb_get($ftp, $local_file, $file, FTP_BINARY);\nwhile ($r == FTP_MOREDATA) {\n    $r = ftp_nb_continue($ftp);\n}\nftp_close($ftp);\necho file_get_contents($local_file);\n?>")).toMatchSnapshot();
  });
});
