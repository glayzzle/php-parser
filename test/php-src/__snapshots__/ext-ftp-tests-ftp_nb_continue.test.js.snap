// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`php-src tests Testing whether ftp_nb_continue() fetches more data 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Include {
        "kind": "include",
        "once": false,
        "require": true,
        "target": String {
          "isDoubleQuote": false,
          "kind": "string",
          "raw": "'server.inc'",
          "unicode": false,
          "value": "server.inc",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "curly": false,
          "kind": "variable",
          "name": "file",
        },
        "operator": "=",
        "right": String {
          "isDoubleQuote": true,
          "kind": "string",
          "raw": "\\"mediumfile.txt\\"",
          "unicode": false,
          "value": "mediumfile.txt",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "curly": false,
          "kind": "variable",
          "name": "ftp",
        },
        "operator": "=",
        "right": Call {
          "arguments": Array [
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": "'127.0.0.1'",
              "unicode": false,
              "value": "127.0.0.1",
            },
            Variable {
              "curly": false,
              "kind": "variable",
              "name": "port",
            },
          ],
          "kind": "call",
          "what": Name {
            "kind": "name",
            "name": "ftp_connect",
            "resolution": "uqn",
          },
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Variable {
            "curly": false,
            "kind": "variable",
            "name": "ftp",
          },
          String {
            "isDoubleQuote": false,
            "kind": "string",
            "raw": "'user'",
            "unicode": false,
            "value": "user",
          },
          String {
            "isDoubleQuote": false,
            "kind": "string",
            "raw": "'pass'",
            "unicode": false,
            "value": "pass",
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "ftp_login",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    If {
      "alternate": null,
      "body": ExpressionStatement {
        "expression": Exit {
          "expression": String {
            "isDoubleQuote": true,
            "kind": "string",
            "raw": "\\"Couldn't connect to the server\\"",
            "unicode": false,
            "value": "Couldn't connect to the server",
          },
          "kind": "exit",
          "useDie": true,
        },
        "kind": "expressionstatement",
      },
      "kind": "if",
      "shortForm": false,
      "test": Unary {
        "kind": "unary",
        "type": "!",
        "what": Variable {
          "curly": false,
          "kind": "variable",
          "name": "ftp",
        },
      },
    },
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "curly": false,
          "kind": "variable",
          "name": "local_file",
        },
        "operator": "=",
        "right": Bin {
          "kind": "bin",
          "left": Bin {
            "kind": "bin",
            "left": Magic {
              "kind": "magic",
              "raw": "__DIR__",
              "value": "__DIR__",
            },
            "right": Name {
              "kind": "name",
              "name": "DIRECTORY_SEPARATOR",
              "resolution": "uqn",
            },
            "type": ".",
          },
          "right": Variable {
            "curly": false,
            "kind": "variable",
            "name": "file",
          },
          "type": ".",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Variable {
            "curly": false,
            "kind": "variable",
            "name": "local_file",
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "touch",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    ExpressionStatement {
      "expression": Assign {
        "kind": "assign",
        "left": Variable {
          "curly": false,
          "kind": "variable",
          "name": "r",
        },
        "operator": "=",
        "right": Call {
          "arguments": Array [
            Variable {
              "curly": false,
              "kind": "variable",
              "name": "ftp",
            },
            Variable {
              "curly": false,
              "kind": "variable",
              "name": "local_file",
            },
            Variable {
              "curly": false,
              "kind": "variable",
              "name": "file",
            },
            Name {
              "kind": "name",
              "name": "FTP_BINARY",
              "resolution": "uqn",
            },
          ],
          "kind": "call",
          "what": Name {
            "kind": "name",
            "name": "ftp_nb_get",
            "resolution": "uqn",
          },
        },
      },
      "kind": "expressionstatement",
    },
    While {
      "body": Block {
        "children": Array [
          ExpressionStatement {
            "expression": Assign {
              "kind": "assign",
              "left": Variable {
                "curly": false,
                "kind": "variable",
                "name": "r",
              },
              "operator": "=",
              "right": Call {
                "arguments": Array [
                  Variable {
                    "curly": false,
                    "kind": "variable",
                    "name": "ftp",
                  },
                ],
                "kind": "call",
                "what": Name {
                  "kind": "name",
                  "name": "ftp_nb_continue",
                  "resolution": "uqn",
                },
              },
            },
            "kind": "expressionstatement",
          },
        ],
        "kind": "block",
      },
      "kind": "while",
      "shortForm": false,
      "test": Bin {
        "kind": "bin",
        "left": Variable {
          "curly": false,
          "kind": "variable",
          "name": "r",
        },
        "right": Name {
          "kind": "name",
          "name": "FTP_MOREDATA",
          "resolution": "uqn",
        },
        "type": "==",
      },
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Variable {
            "curly": false,
            "kind": "variable",
            "name": "ftp",
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "ftp_close",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
    Echo {
      "expressions": Array [
        Call {
          "arguments": Array [
            Variable {
              "curly": false,
              "kind": "variable",
              "name": "local_file",
            },
          ],
          "kind": "call",
          "what": Name {
            "kind": "name",
            "name": "file_get_contents",
            "resolution": "uqn",
          },
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
